### PR TITLE
#4044 #4043 Limit the information exported to the lines selected in the list of the directory + Add custom export on EventParticipant page

### LINKED ISSUE #4043: Personalized export of people participating in the event
### Situation Description

To date, lack of a personalized export for the part participating in the event. 

### Feature Description

This personalized export would be: 

- Configurable and modifiable by all users (each user configures their own export) like personalized exports in cases or contacts for example 

- Configurable by a supervisor knowing that this configuration, once recorded, appears for all users and cannot be modified by them (formatting of a personalized export allowing to standardize the export to be used to communicate information to an institutional partner for example)

### Additional Information
@ftavin @olivierleuci @bernardsilenou  @emond

### LINKED ISSUE #4044: Limit the information exported to  the lines selected in the list of the directory
### Situation Description

To date, the basic export makes it possible to export the lines displayed in the various directories, including those of the participants in the event;
It would be useful to be able to manually select the lines to include in the export.

### Feature Description

The detailed export created by issue #4043 would have an option to limit exports to all selected lines in the directory. In the event that no row is selected; then the export will concern all the lines displayed on the screen
### Additional Information

@ftavin @bernardsilenou @olivierleuci @emond

### PR DESCRIPTION
**Pull request by mathieu-petit-atol**
Closes #4044,
Closes #4043

### CODE REVIEW

**Comment by ftavin** [comment]
@MateStrysewske I'm not sure you had the information but "mpe-a" (Mathieu) is a developper from my team, he is joining the SORMAS effort while Alexandre is shifting on another project for a while

File: sormas-ui/src/main/java/de/symeda/sormas/ui/caze/CasesView.java [hunk 1]
```diff
@@ -632,6 +653,12 @@ public HorizontalLayout createStatusFilterBar() {
 
 					final List<MenuBarHelper.MenuBarItem> menuBarItems = new ArrayList<>();
 
+					menuBarItems.add(
```
Comment (MateStrysewske) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/caze/CasesView.java:656]: I think this should be removed

File: sormas-ui/src/main/java/de/symeda/sormas/ui/utils/GridExportWithSelectedItemStreamResource.java [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+import de.symeda.sormas.api.caze.AgeAndBirthDateDto;
+import de.symeda.sormas.api.i18n.I18nProperties;
+import de.symeda.sormas.api.i18n.Strings;
+import de.symeda.sormas.api.person.PersonHelper;
+import de.symeda.sormas.api.utils.CSVUtils;
+import de.symeda.sormas.api.utils.YesNoUnknown;
+import de.symeda.sormas.ui.caze.AbstractCaseGrid;
+
+@SuppressWarnings("serial")
+public class GridExportWithSelectedItemStreamResource extends StreamResource {
```
Comment (MateStrysewske) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/utils/GridExportWithSelectedItemStreamResource.java:58]: Rename to `GridExportWithSelectedItemsStreamResource`

**Review by MateStrysewske — CHANGES_REQUESTED** [review]
I'd like you to get rid of the new captions that you added to the properties again; they should be handled either by the Location captions (e.g. house number, postal code, etc.) or by the general captions without a prefix (first name, last name, region, district, ...) - I think it should already work like this for the other exports

**Comment by MateStrysewske** [comment]
@ftavin @mpe-a I'm off work for the next days and will be back on Tuesday. I'll continue reviewing pull requests then!

**Review by MateStrysewske — CHANGES_REQUESTED** [review]
* Remove the newly added captions; they should be handled by the captions that are already there (either with the Location prefix, e.g. street, or without a prefix, e.g. first name); you probably need to add fallbacks to the code that creates the captions for the event participants export file; there probably has to already be such a mechanism in place for e.g. the case export because that also contains first and last name

**Review by MateStrysewske — CHANGES_REQUESTED** [review]
* Event actions export (Events -> Toggle to "Actions" on top -> Export) is not working anymore
* Basic event participant export is broken (has no file extension) - not sure whether this is because of changes in this PR, but it would be great if you could fix it

The new logic is not implemented/working for
* Basic task export
* Case follow-up visits
I think it would be good to have it for these exports as well to be consistent throughout the whole application

**Comment by mathieu-petit-atol** [comment]
> The new logic is not implemented/working for
> 
> * Basic task export
> * Case follow-up visits
>   I think it would be good to have it for these exports as well to be consistent throughout the whole application

I did it for the task directory but for the case follow-up visits it would seem that they are no bulk edit option on this page (and they are no export button too).

**Comment by MateStrysewske** [comment]
> I did it for the task directory but for the case follow-up visits it would seem that there are no bulk edit option on this page (and there are no export button too).
Have you tried to log in as a user with the user roles admin + national user? I recommend that for most development work because it allows you to see all the functionality in the system.

Another option is that I didn't make clear what I meant - I'm not talking about the general follow-up overview of all cases, but about the follow-up tab inside a single case. The grid in there is by default in bulk edit mode and has an export functionality, similar to the follow-up tab inside a single contact.

**Comment by MateStrysewske** [comment]
Sorry for the amount of merge conflicts today, there's a lot of activity because we approach the end of the sprint. Can you resolve them please?

**Review by MateStrysewske — APPROVED** [review]

**Pull request merged**